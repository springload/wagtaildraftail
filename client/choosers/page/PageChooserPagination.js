import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  totalPages: PropTypes.number.isRequired,
  pageNumber: PropTypes.number,
  onChangePage: PropTypes.func.isRequired,
};

const defaultProps = {
  pageNumber: 0,
};

class PageChooserPagination extends React.Component {
  renderPrev() {
    const { pageNumber, onChangePage } = this.props;
    const hasPrev = pageNumber !== 1;

    if (hasPrev) {
      const onClickPrevious = (e) => {
        onChangePage(pageNumber - 1);
        e.preventDefault();
      };

      return (
        <li className="prev">
          <a
            onClick={onClickPrevious}
            href="#"
            className="icon icon-arrow-left navigate-pages"
          >
            Previous
          </a>
        </li>
      );
    } else {
      return (
        <li className="prev" />
      );
    }
  }

  renderNext() {
    const { pageNumber, onChangePage, totalPages } = this.props;
    const hasNext = pageNumber < totalPages;

    if (hasNext) {
      const onClickNext = (e) => {
        onChangePage(pageNumber + 1);
        e.preventDefault();
      };

      return (
        <li className="next">
          <a
            onClick={onClickNext}
            href="#"
            className="icon icon-arrow-right-after navigate-pages"
          >
            Next
          </a>
        </li>
      );
    } else {
      return (
        <li className="next" />
      );
    }
  }

  render() {
    const { totalPages, pageNumber } = this.props;

    return (
      <div className="pagination">
        {totalPages > 1 ? (
          <div>
            <p>
              {`Page ${pageNumber} of ${totalPages}.`}
            </p>
            <ul>
              {this.renderPrev()}
              {this.renderNext()}
            </ul>
          </div>
        ) : null}
      </div>
    );
  }
}

PageChooserPagination.propTypes = propTypes;
PageChooserPagination.defaultProps = defaultProps;

export default PageChooserPagination;
