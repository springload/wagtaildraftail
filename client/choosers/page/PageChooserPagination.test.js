import React from 'react';
import { shallow } from 'enzyme';
import PageChooserPagination from './PageChooserPagination';

describe('PageChooserPagination', () => {
  it('renders', () => {
    expect(shallow((
      <PageChooserPagination
        totalPages={0}
        onChangePage={() => {}}
      />
    ))).toMatchSnapshot();
  });

  it('has pages', () => {
    expect(shallow((
      <PageChooserPagination
        totalPages={2}
        onChangePage={() => {}}
      />
    ))).toMatchSnapshot();
  });
});
