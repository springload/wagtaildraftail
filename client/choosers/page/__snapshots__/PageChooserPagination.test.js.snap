// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageChooserPagination has pages 1`] = `
<div
  className="pagination"
>
  <div>
    <p>
      Page 0 of 2.
    </p>
    <ul>
      <li
        className="prev"
      >
        <a
          className="icon icon-arrow-left navigate-pages"
          href="#"
          onClick={[Function]}
        >
          Previous
        </a>
      </li>
      <li
        className="next"
      >
        <a
          className="icon icon-arrow-right-after navigate-pages"
          href="#"
          onClick={[Function]}
        >
          Next
        </a>
      </li>
    </ul>
  </div>
</div>
`;

exports[`PageChooserPagination renders 1`] = `
<div
  className="pagination"
/>
`;
